scripts/docker_run

```diff
@@ -19,16 +21,23 @@ docker build \
   --tag=oak \
   .
 
-# Keep this in sync with /scripts/run_examples .
-docker run \
-  --interactive \
-  --tty \
-  --user="$DOCKER_UID:$DOCKER_GID" \
-  --env="USER=$DOCKER_USER" \
-  --volume="$PWD/bazel-cache:/.cache/bazel" \
-  --volume="$PWD/cargo-cache:/usr/local/cargo/registry" \
-  --volume="$PWD:/opt/my-project" \
-  --workdir=/opt/my-project \
-  --network=host \
-  oak:latest \
-  "$@"
+docker_run_flags=(
+  "--interactive"
+  "--tty"
+  "--user=$DOCKER_UID:$DOCKER_GID"
+  "--env=USER=$DOCKER_USER"
+  "--volume=$PWD/bazel-cache:/.cache/bazel"
+  "--volume=$PWD/cargo-cache:/usr/local/cargo/registry"
+  "--volume=$PWD:/opt/my-project"
+  "--workdir=/opt/my-project"
+  "--network=host"
+)
+
+if [[ "$1" == "--detach" ]]; then
+  docker_run_flags+=("--detach")
+  docker run "${docker_run_flags[@]}" oak:latest "${@:2}"
+  # Run `docker ps --quiet --latest` in caller to get reference to detached container.
```

Comment:
Could we make it return the id, like `docker run --detach` itself?

Reply:
Right, that would be awesome, my first implementation did that. In order to do so the previous build command would need to be silenced `docker build ... > /dev/null`. I figured that loosing the build log costs more than not getting the container id returned from the `docker_run` script. 

See discussion under https://github.com/project-oak/oak/pull/112#discussion_r294209897

Reply:
Got it, thanks! I still think we could redirect all output to stderr and return the id on stdout perhaps. But we can look into it separately.

Reply:
That's right, good idea! I will look into it.